blogs/dns_tracking/dns_tracking.md

```diff
@@ -0,0 +1,136 @@
+to correlate the query or response for specific DNS transaction, and then record the
+elapsed time to compute DNS latency. Further, DNS network flows are enriched
+to include fields, viz. DNS Id, DNS latency and DNS response codes to help
+build graphs with aggregated DNS statistics and to help filtering on specific
+fields in the Network Observability console.
+
+## Potential Use Cases
+
+DNS tracking with eBPF tracepoint hooks can serve various purposes:
+
+- Network Monitoring: Gain insights into DNS queries and responses,
+helping network administrators identify unusual patterns,
+potential bottlenecks, or performance issues.
+
+- Security Analysis: Detect suspicious DNS activities, such as domain
+name generation algorithms (DGA) used by malware,
+or identify unauthorized DNS resolutions that might indicate a security breach.
+
+- Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps,
+tracking latency, and identifying misconfigurations.
+
+## How to enable DNS tracking
+
+By default DNS tracking is disabled because it requires
+`privileged` access, to enable this feature we need to create a flow
+collector object with the following fields enabled in eBPF config
+section
+
+```yaml
+apiVersion: flows.netobserv.io/v1beta1
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  agent:
+    type: EBPF
+    ebpf:
+      privileged: true
+      features:
+        - DnsTacking
+```
+
+## A quick tour in the UI
+
+Once `DnsTacking` feature enabled, the Console plugin will automatically
+adapt to provide additional filters and show informations across views.
+
+Open your OCP Console and move to
+`Administrator view` -> `Observe` -> `Network Traffic` page as usual.
+
+Three new filters, `DNS Id`, `DNS Latency` and `DNS Response Code` will be available
+in the common section:
+
+![dns filters](./images/dns_filters.png)
+
+The first one will allow you to filter on a specific DNS Id to correlate
+with your query.
+
+![dns id](./images/dns_id.png)
+
+The second one helps to identify potential performance issues by looking at DNS
+resolution latency.
+
+![dns latency more than](./images/dns_latency_more_than.png)
+
+The third filter surfaces DNS response codes, which can help detect errors or
+unauthorized resolutions.
+
+![dns rcode](./images/dns_response_code.png)
+
+### Overview
+
+New graphs are introduced in the `advanced options` -> `manage panels` popup:
+
+![advanced options 1](./images/advanced_options1.png)
+
+- Top 5 average DNS latencies
+- Top 5 DNS response code
+- Top 5 DNS response code stacked with total
+
+![dns graphs 1](./images/dns_graphs1.png)
+![dns graphs 2](./images/dns_graphs2.png)
+
+### Traffic flows
+
+The table view adds the new DNS columns `Id`, `Latency` and
+`Response code`, which are available from the `advanced options` -> `manage columns` popup.
+
+![advanced options 2](./images/advanced_options2.png)
+
+The DNS flows display this information in both the table and the side panel:
```

Comment:
Note that this is where you would find the DNS Id to use in the DNS filters query above.